### PR TITLE
SNOW-1445360: Add Support for Series.str.get

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1445360

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   SNOW-1445360: Add Support for Series.str.get.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -12269,8 +12269,55 @@ def output_col(col_name: ColumnOrName) -> SnowparkColumn:
...
+                    ),
+                )
+            else:
+                assert i >= 0
+                # Index is relative to the beginning boundary.
+                # If it falls after the end boundary, Null is returned.
+                new_col = iff(
+                    pandas_lit(i) >= col_len_exp,
+                    pandas_lit(None),
+                    substring(col(col_name), pandas_lit(i + 1), pandas_lit(1)),
```
Comment (sfc-gh-jkew) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:12313]: Nit: You might want to make the transition from zero-based to one-based clearer. It always confuses me when reading the code.

**Review by sfc-gh-jkew — APPROVED** [review]
Looks good. Only confusion was the transition to 1-based.

File: docs/source/modin/supported/series_str_supported.rst [hunk 2]
```diff
@@ -44,7 +44,8 @@ the method in the left column.
 +-----------------------------+---------------------------------+----------------------------------------------------+
 | ``fullmatch``               | N                               |                                                    |
 +-----------------------------+---------------------------------+----------------------------------------------------+
-| ``get``                     | N                               |                                                    |
+| ``get``                     | P                               | ``N`` if the `i` parameter is set to a non-int     |
```
Comment (sfc-gh-nkumar) [inline docs/source/modin/supported/series_str_supported.rst:47]: Current implementation also doesn't support lists and dict. Shall we mention that here?

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 3]
```diff
@@ -12269,8 +12269,55 @@ def output_col(col_name: ColumnOrName) -> SnowparkColumn:
         )
         return SnowflakeQueryCompiler(new_internal_frame)
 
-    def str_get(self, i: int) -> None:
-        ErrorMessage.method_not_implemented_error("get", "Series.str")
+    def str_get(self, i: int) -> "SnowflakeQueryCompiler":
+        """
+        Extract element from each component at specified position or with specified key.
+
+        Extract element from lists, tuples, dict, or strings in each element in the Series/Index.
```
Comment (sfc-gh-nkumar) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:12276]: > lists, tuples, dict, or strings

This only supports extraction from string at the moment?

**Review by sfc-gh-nkumar — COMMENTED** [review]
String implementation looks good. Pandas also supports this for lists and dicts. Please clarify this in documentation that it's not yet supported in snowpark pandas.

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:12315]: This applies to all string methods. I added a comment to clarify it here though.

File: docs/source/modin/supported/series_str_supported.rst (hunk 2 above)
Comment (sfc-gh-helmeleegy) [reply docs/source/modin/supported/series_str_supported.rst:47]: Good point. Updated the docs.

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 3 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:12276]: That's right. Updated docs.

**Review by sfc-gh-nkumar — APPROVED** [review]

**Pull request merged**